RoosterPlanner.Data/Repositories/ShiftsRepository.cs

```diff
@@ -61,5 +100,33 @@ public Task<Shift> GetShiftAsync(Guid shiftId)
                 .Include(s => s.Project)
                 .Where(s => s.Id == shiftId).FirstOrDefaultAsync();
         }
+
+        public async Task<List<Shift>> GetByProjectUserAndDateAsync(Guid projectId, Guid userId, DateTime date)
+        {
+            if (projectId == Guid.Empty || userId == Guid.Empty)
+                return null;
+            List<Shift> listOfShifts = await EntitySet
+                .AsNoTracking()
+                .Include(s => s.Project)
+                .Include(s => s.Task)
+                .Include(s => s.Availabilities)
+                .ThenInclude(a => a.Participation)
+                .Where(s => s.ProjectId == projectId && s.Date == date)
+                .ToListAsync();
+            listOfShifts.ForEach(s =>
```

Comment:
Als de lijst van objecten wat langer is en volgorde niet relevant, kan je ook Parrallel.ForEach() gebruiken